src/grain/ledger.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import {type NodeAddressT, NodeAddress} from "../core/graph";
+import stringify from "json-stable-stringify";
+import * as NullUtil from "../util/null";
+import {
+  type Grain,
+  format as formatGrain,
+  ZERO,
+  DECIMAL_PRECISION,
+} from "./grain";
+import {type DistributionV1} from "./distribution";
+
+/**
+ * Models a transfer of grain from a sender to recipient.
+ * A grainholder may transfer grain to themself (which is a no-op).
+ */
+export type TransferV1 = {|
+  +type: "TRANSFER",
+  +version: number,
+  +sender: NodeAddressT,
+  +recipient: NodeAddressT,
+  +amount: Grain,
+  +timestampMs: number,
```

Comment:
I think it would be beneficial to have an optional field for arbitrary data / memo. Will be helpful if we want to add a related link or on-chain txID or any other message for context on what that transfer was for. Can just be a string.

Reply:
Thinking of AraCred, one way to do the on-chain bridge would be to have an "Ethereum" node address, and when someone wants to take some Grain on chain, it would just be modelled as a Transfer from their grain balance to the "Ethereum" node's grain balance. That data field would then contain the txID of minting that grain on chain.

The grain is then free to travel anywhere on chain, be bought and sold, etc, and then whoever wants to bring it back in the grain ledger, it would just be modelled as a Transfer from the "Ethereum" node address to that users node address, with the data field containing the txID of burning that token on chain.

Reply:
Yeah, I agree. This would serve the same need as [references](https://github.com/sourcecred/cred/blob/aed8b172d3e3b4fa7a61a6cad13100513e5ca9ff/transfers.json#L32) in our current ad-hoc system